<!-- -->
<launch>

    <!-- Setting pose origin -->
    <arg name="topic_pose_stamped" default="/gnss_pose"/> <!-- ndt_pose | icp_pose | gnss_pose -->

    <!-- Setup -->
    <include file="$(find runtime_manager)/scripts/setup_tf.launch">
        <arg name="x" value="-0.77"/><!---0.77-->
        <arg name="y" value="0.0"/>
        <arg name="z" value="1.07"/><!-- 1.07-->
        <arg name="yaw" value="0.0"/>
        <arg name="pitch" value="0.0"/>
        <arg name="roll" value="0.0"/>
        <arg name="frame_id" value="base_link"/>
        <arg name="child_frame_id" value="/velodyne"/>
        <arg name="period_in_ms" value="10"/>
    </include>

    <!-- Vehicle Model -->
    <include file="$(find model_publisher)/launch/vehicle_model.launch">
        <arg name="model_path" value="$(find adeye)/../../../../AD-EYE/Data/URDF_Actor_Models/BMW_X5/default.urdf"/>
    </include>

    <!-- Points Downsampler -->
    <node pkg="rostopic" type="rostopic" name="config_voxel_grid_filter_rostopic"
          args="pub -l /config/voxel_grid_filter autoware_config_msgs/ConfigVoxelGridFilter
        '{voxel_leaf_size: 0.2 , measurement_range: 200.0}' "
    />
    <include file="$(find points_downsampler)/launch/points_downsample.launch"/>

    <!-- ndt_matching --> <!-- METHOD TIPE: pcl_generic=0, pcl_anh=1, pcl_anh_gpu=2, pcl_openmp=3 -->
    <!--<include file="$(find lidar_localizer)/launch/ndt_matching.launch">
      <arg name="method_type" value="0" />
      <arg name="use_gnss" value="1" />
      <arg name="use_odom" value="false" />--> <!--THIS HELPS EVEN WHEN YOU DONT PUBLISH ODOM IMFORMATION-->
    <!--<arg name="use_imu" value="false" />
    <arg name="imu_upside_down" value="false" />
    <arg name="imu_topic" value="imu_raw" />
    <arg name="queue_size" value="1" />
    <arg name="offset" value="linear" />
    <arg name="get_height" value="false" />
    <arg name="use_local_transform" value="true" />
    <arg name="sync" value="false" />
  </include>-->

    <!-- icp_matching -->
    <!--<include file="$(find lidar_localizer)/launch/icp_matching.launch">
    </include>-->

    <!-- GPS to base_link: Use only when not using the ndt_matching -->
    <node pkg="tf" type="static_transform_publisher" name="gps_to_base_link"
          args="0.0 0.0 0.0 0.0 0.0 0.0 1.0 gps base_link 10"/>

    <!-- Pose Connect -->
    <node pkg="topic_tools" type="relay" name="pose_relay" output="log" args="$(arg topic_pose_stamped) /current_pose"/>

</launch>
